src/uct/ib/base/ib_device.c

```diff
@@ -913,50 +913,44 @@ uct_ib_device_roce_gid_to_sockaddr(sa_family_t af, const void *gid,
 }
 
 static ucs_status_t
-uct_ib_device_parse_subnet_list(const ucs_config_allow_list_t *subnet_strs,
-                                uct_ib_device_subnet_array_t *subnets)
+uct_ib_device_parse_subnet_filter(const ucs_config_allow_list_t *subnet_strs,
+                                  uct_ib_device_subnet_array_t *subnets)
 {
-    static const unsigned subnet_num_elems = 2;
-    static const unsigned sscanf_res_ok    = 1;
-    char *address_str, *mask_str, *subnet_str_dup;
+    static const size_t max_subnet_str_len = 60;
+    char *address_str, *mask_str;
     char **subnet_str;
+    char subnet_str_dup[max_subnet_str_len];
     uct_ib_device_subnet_t *subnet;
     ucs_status_t status;
-    size_t subnet_str_len;
 
-    if (!uct_ib_device_roce_subnet_enabled(subnet_strs->mode)) {
+    if (subnet_strs->mode == UCS_CONFIG_ALLOW_LIST_ALLOW_ALL) {
         return UCS_OK;
     }
 
     ucs_carray_for_each(subnet_str, subnet_strs->array.names,
                         subnet_strs->array.count) {
-        /* Duplicate subnet_str to prevent ucs_string_split from overwriting
-         * it */
-        subnet_str_len = strlen(*subnet_str) + 1; /* Extra space for '\0' */
-        subnet_str_dup = ucs_alloca(subnet_str_len);
-        ucs_strncpy_zero(subnet_str_dup, *subnet_str, subnet_str_len);
+        ucs_strncpy_zero(subnet_str_dup, *subnet_str, max_subnet_str_len);
```

Comment:
can just use ucs_strncpy - no need to fill the buffer with 0 after the regulat \0 at the end of the string